int64[2] states